db-tests/src/test/java/gov/nasa/jpl/aerie/database/ExternalEventTests.java

```diff
@@ -0,0 +1,2200 @@
+        insertExternalEvent(e);
+        insertExternalEvent(b1);
+        insertExternalEvent(b2);
+
+        // verify the expected keys
+        final var results = getDerivedEvents();
+
+        // both ranges should only have a single element and be fully present
+        final List<DerivedEvent> expectedResults = List.of(
+            new DerivedEvent(
+                "a",
+                SOURCE_TYPE,
+                "A",
+                DERIVATION_GROUP,
+                "2024-01-01 00:25:00+00",
+                "00:10:00",
+                "{[\"2024-01-01 00:00:00+00\",\"2024-01-01 00:30:00+00\")}",
+                "2024-01-01 00:00:00+00"
+            ),
+            new DerivedEvent(
+                "b1",
+                SOURCE_TYPE,
+                "B",
+                DERIVATION_GROUP,
+                "2024-01-01 00:30:00+00",
+                "00:10:00",
+                "{[\"2024-01-01 00:30:00+00\",\"2024-01-01 01:30:00+00\")}",
+                "2024-01-02 00:00:00+00"
+            ),
+            new DerivedEvent(
+                "b2",
+                SOURCE_TYPE,
+                "B",
+                DERIVATION_GROUP,
+                "2024-01-01 00:45:00+00",
+                "00:10:00",
+                "{[\"2024-01-01 00:30:00+00\",\"2024-01-01 01:30:00+00\")}",
+                "2024-01-02 00:00:00+00"
+            )
+        );
+        assertEquals(expectedResults.size(), results.size());
+        assertTrue(results.containsAll(expectedResults));
+      }
+
+      ////////////////////////// RULE 3 //////////////////////////
+      /**
+       * Test a basic case of rule 3, wherein a superseding source (regardless of empty space) nullifies anything in A
+       *    based on the _start time_ of the event occurring during the range of a newer source.
+       *
+       *    A:    +a+aaaaa
+       *    B:  b+bb++++
+       *    (only b's should be in result)
+       */
+      @Test
+      void rule3_basic() throws SQLException {
+        // insert the event(s) (and their source(s))
+        ExternalSource A = new ExternalSource(
+            "A",
+            SOURCE_TYPE,
+            DERIVATION_GROUP,
+            "2024-01-01T00:00:00Z",
+            "2024-01-01T00:30:00Z",
+            "2024-01-01T01:30:00Z",
+            CREATED_AT
+        );
+        ExternalSource B = new ExternalSource(
+            "B",
+            SOURCE_TYPE,
+            DERIVATION_GROUP,
+            "2024-01-02T00:00:00Z",
+            "2024-01-01T00:00:00Z",
+            "2024-01-01T01:00:00Z",
+            CREATED_AT
+        );
+
+        ExternalEvent e1 = new ExternalEvent(
+            "a1",
+            "2024-01-01T00:40:00Z",
+            "00:10:00",
+            A
+        ); // negated by B, very clearly
+        ExternalEvent e2 = new ExternalEvent(
+            "a2",
+            "2024-01-01T00:55:00Z",
+            "00:35:00",
+            A
+        ); // even empty space in B neg should negate
```

Comment:
Please move these comments to be on the line above the code it applies to and not as a trailing thought at the end of a line of code.